core/dbt/task/test.py

```diff
@@ -213,16 +220,18 @@ def execute_unit_test(
                 "Invalid materialization context generated, missing config: {}".format(context)
             )
 
-        # generate materialization macro
-        macro_func = MacroGenerator(materialization_macro, context)
-        # execute materialization macro
         try:
+            # generate materialization macro
+            macro_func = MacroGenerator(materialization_macro, context)
```

Comment:
Is there concern that generating `macro_func` would throw an exception that we want to catch (hence moving it inside the `try`)? If not, I'd leave this where it is. We want to minimize what runs inside of a `try`.

Reply:
I don't know of any outright concern about the intilizer having it in the try more so because we only call it the one time each time so it seemed like it should stay as close together as possible, if others feel differently can try pulling it out.

Reply:
pulled it out